dashboard/test/ui/features/acquisition_products/pd/teacher_application.feature

```diff
@@ -203,4 +203,161 @@ Scenario: Teacher saves, re-opens, and submits an application
   Then I wait until element "h1" contains text "Thank you for submitting your application!"
   Then I see no difference for "Confirmation after submitting previously saved application"
   Then I close my eyes
```

Comment:
Don't forget to add `Then I close my eyes` to your new tests.

Reply:
Good call. This raises a question I have –– do we have norms around what is an eyes test and what is a UI test? Do we want snapshots for each page of each variation? Or just UI tests? Is it weird to test more-or-less the same functionality via an eyes test and a UI test?

Reply:
These are all good questions that I don't have great answers to. Eyes tests are more expensive to run (and update) than UI tests, so my understanding has always been to default to non-eyes UI tests. The question becomes what are you trying to test? Are you trying to test that a set of clicks works as expected? Or are you testing that something actually looks correct? The first should be a UI test (if possible) and the second an eyes test.

As for snapshots of each page of each variation -- I'd probably lean towards not taking snapshots of the same page multiple times, but could definitely understand an argument against that.

In this specific case, as the existing tests are already eyes tests, I think it's just up to preference. Would it be easier to maintain if all these tests were mostly the same? Then make them either all eyes tests or not. Do you think it's okay to mix and match? Then maybe one or two tests being eyes tests is enough and the rest could be UI tests.

Reply:
You can mix and match in the same file! You just need to move the `@eyes` tag to just above each scenario, like in https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/ui/features/teacher_tools/level_types/multiple_choice_contained_levels.feature.

Reply:
Ah, excellent! What do you think of that, Turner?

Reply:
Mix and matching sounds good! I'm a bit confused as to what the tests do if @eyes isn't pre-appended? Or like what the difference is?

Reply:
Good question –– they both use Cucumber to run the tests, which is nice because the steps can usually stay the same. But eyes test includes a visual snapshot of the test when you write "And I see no difference for ..." and compares it to the last snapshot. Those are what you see in Applitools (e.g. https://eyes.applitools.com/app/test-results/00000251734988317245/00000251734987739400/steps/1?accountId=mG4DoRfv0ntEgI3XIYBulrKxlZqrHtGwbP3V103ojvLFU110&mode=step-editor)

When a UI test fails, you'll see the Cucumber log like https://cucumber-logs.s3.amazonaws.com/circle/23699/Chrome_acquisition_products_pd_teacher_application_eyes_output.html?versionId=oVSjLDwSlDEZ9q.vsHAMu7QcRqOx_75a. << That's an eyes test, but it failed on a step in Cucumber. When an eyes test fails, it's possible the Cucumber logs may have run without problems, but the snapshot in Applitools looks different, causing the test to fail.

Reply:
Will do, thank you both!